integration-tests/runtime-attach/runtime-attach-test/src/test/java/co/elastic/apm/testapp/RuntimeAttachTest.java

```diff
@@ -88,41 +103,137 @@ void after() {
         }
         forkedJvms.clear();
         jmxUrl = null;
+
+        assertThat(Path.of(System.getProperty("java.io.tmpdir"), ELASTICAPM_PROPERTIES))
+            .doesNotExist();
+
+    }
+
+    @Test
+    void testSimpleAttachCli() {
+        testSimpleAttach(false);
     }
 
     @Test
-    void cliAttach() {
-        ProcessHandle appJvm = startAppForkedJvm(false);
+    void testSimpleAttachSelfAttach() {
+        testSimpleAttach(true);
+    }
+
+    private void testSimpleAttach(boolean selfAttach) {
+        ProcessHandle appJvm = startAppForkedJvm(selfAttach, null);
         long pid = appJvm.pid();
 
         waitForJmxRegistration(pid);
 
-        await().until(() -> getWorkUnitCount(false) > 0);
-        assertThat(getWorkUnitCount(true)).isEqualTo(0);
+        await("wait for application to start work")
+            .until(() -> getWorkUnitCount(false) > 0);
+
+        if (!selfAttach) {
+            assertThat(getWorkUnitCount(true))
+                .describedAs("instrumentation should not start before using CLI attacher")
+                .isEqualTo(0);
+
+            startAttacherForkedJvm(
+                "--include-pid",
+                Long.toString(pid),
+                "--config",
+                "disable_send=true",
+                "application_packages=co.elastic.apm.testapp",
+                "cloud_provider=NONE");
+        }
 
-        startAttacherForkedJvm(
-            "--include-pid",
-            Long.toString(pid),
-            "--config",
-            "disable_send=true",
-            "application_packages=co.elastic.apm.testapp",
-            "cloud_provider=NONE");
+        waitForInstrumentedWork();
+    }
+
+    @Test
+    void testExternalConfigurationCli(@TempDir File tmp) throws IOException {
+        // only testing with cli-attach for simplicity as it's easier to provide custom parameters
+        // should behave in a similar way with
```

Comment:
should behave in a similar way with what?

Reply:
similar was as using the Attacher API.